### PR TITLE
Always include the very first and last points for fills

### PR DESCRIPTION
**Pull request by etpinard**
... during the line-decimation algo - fixes https://github.com/plotly/plotly.js/issues/3147

cc @plotly/plotly_js

### CODE REVIEW

File: test/image/mocks/ultra_zoom_fill.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "data": [
```
Comment (etpinard) [inline test/image/mocks/ultra_zoom_fill.json]: which looks like:

image


on `master`.

**Comment by antoinerg** [comment]
Thanks for the fix @etpinard !

:dancer:

File: src/traces/scatter/line_points.js [hunk 2]
```diff
@@ -387,7 +391,9 @@ module.exports = function linePoints(d, opts) {
 
             clusterRefDist = ptDist(clusterHighPt, clusterStartPt);
 
-            if(clusterRefDist < getTolerance(clusterHighPt, nextPt) * minTolerance) continue;
+            // #3147 - always include the very first and last points for fills
+            if(!(fill && (pti === 0 || len - pti === 1)) &&
```
Comment (alexcjohnson) [inline src/traces/scatter/line_points.js]: I've never seen `pti === len - 1` written quite that way 🤔

Comment (etpinard) [reply src/traces/scatter/line_points.js]: yeah, that _was_ weird, fixed in https://github.com/plotly/plotly.js/pull/3696/commits/9e8c3e9783860cc741bb3fb9bd0e9a1aa754ab44

**Comment by archmoj** [comment]
:dancer:

**Pull request merged**